lib/date.js

```diff
@@ -98,8 +98,18 @@ module.exports = (key, options) => {
         label: label === labelKey ? v.label : label
       });
     }));
+    const legendKey = `fields.${key}.legend`;
+    let legend = req.translate(legendKey);
+    if (legend === legendKey) {
+      legend = null;
+    }
+    const hintKey = `fields.${key}.hint`;
+    let hint = req.translate(hintKey);
+    if (hint === hintKey) {
+      hint = null;
+    }
```

Comment:
Just to save on repetition, you could extract the conditional translation logic to a helper function:

```js
const conditionalTranslate = (key, translate) => {
  let result = translate(key);
  if (result === key) {
    result = null;
  }
  return result;
};

// then use it like:
const legend = conditionalTranslate(`fields.${key}.legend`, req.translate);
const hint = conditionalTranslate(`fields.${key}.hint`, req.translate);
```